### PR TITLE
Raise an exception when the scheduling role cannot log in

### PR DESCRIPTION
**Pull request by avaczi**
And add line breaks to the very long line

### CODE REVIEW

File: postgres-appliance/post_init.sh [hunk 1]
```diff
@@ -6,7 +6,11 @@ CREATE ROLE robot_zmon;
...
 ALTER TABLE cron.job ALTER COLUMN nodename SET DEFAULT '/var/run/postgresql';
-ALTER POLICY cron_job_policy ON cron.job USING (username = current_user OR pg_has_role(current_user, 'admin', 'MEMBER') AND pg_has_role(username, 'admin', 'MEMBER') AND NOT EXISTS(SELECT 1 FROM pg_roles WHERE rolname = username AND rolsuper));
+ALTER POLICY cron_job_policy ON cron.job USING (username = current_user OR
+    (pg_has_role(current_user, 'admin', 'MEMBER')
+    AND pg_has_role(username, 'admin', 'MEMBER')
+    AND NOT EXISTS(SELECT 1 FROM pg_roles WHERE rolname = username AND rolsuper)
+    ));
 REVOKE SELECT ON cron.job FROM public;
 GRANT SELECT ON cron.job TO admin;
 GRANT UPDATE (database) ON cron.job TO admin;
```
Comment (CyberDem0n) [inline postgres-appliance/post_init.sh]: GRANT UPDATE (database, nodename) ON cron.job TO admin;

File: postgres-appliance/post_init.sh [hunk 2]
```diff
@@ -6,7 +6,11 @@ CREATE ROLE robot_zmon;
 
 CREATE EXTENSION pg_cron;
 ALTER TABLE cron.job ALTER COLUMN nodename SET DEFAULT '/var/run/postgresql';
```
Comment (CyberDem0n) [inline postgres-appliance/post_init.sh]: please remove it

File: postgres-appliance/post_init.sh [hunk 3]
```diff
@@ -18,6 +22,10 @@ AS \$function\$
 DECLARE
     l_jobid bigint;
 BEGIN
+    IF NOT (SELECT rolcanlogin FROM pg_roles WHERE rolname = current_user)
+    THEN RAISE 'You cannot create a job using a role that cannot log in';
+    END IF;
+
     SELECT schedule INTO l_jobid FROM cron.schedule(p_schedule, p_command);
     UPDATE cron.job SET database = p_database WHERE jobid = l_jobid;
```
Comment (CyberDem0n) [inline postgres-appliance/post_init.sh]: UPDATE cron.job SET nodename='', database = p_database WHERE jobid = l_jobid;

**Comment by CyberDem0n** [comment]
:+1:

**Comment by avaczi** [comment]
:+1:

**Pull request merged**